.travis.yml

```diff
@@ -21,70 +21,123 @@ before_install:
 
 install: true
 
 jobs:
   include:
     - stage: "Verify vendor"
+      arch: amd64
       script: make verify
-      name: "check if the vendor is up to date"
+      name: "check if the vendor is up to date on amd64"
+    - script: make verify
+      arch: arm64
+      name: "check if the vendor is up to date on arm64"
 
     - stage: "Test lint"
       script: make cloud_lint
-      name: "cloud_lint"     # names the eighth Tests stage job
+      name: "cloud_lint"
     - script: make bluetoothdevice_lint
       name: "bluetooth_device_lint"
     - script: make keadm_lint
       name: "keadm_lint"
     - script: make edge_lint
       name: "edge_lint"
 
-    - stage: "Test Build"                # naming the Tests stage
-      script: make 
-      name: "builds cloud and edge components"            # names the first Tests stage job
+    - stage: "Test Build"
+      script: make
+      arch: amd64
+      name: "builds cloud and edge components on amd64"
+    - script: make
+      arch: arm64
+      name: "builds cloud and edge components on arm64"
     - script: make bluetoothdevice
-      name: "build bluetooth_device"
+      arch: amd64
+      name: "build bluetooth_device on amd64"
+    - script: make bluetoothdevice
+      arch: arm64
+      name: "build bluetooth_device on arm64"
+    - script: make edge_small_build
+      arch: amd64
+      name: "build small edge version on amd64"
     - script: make edge_small_build
-      name: "build small edge version"
+      arch: arm64
+      name: "build small edge version on arm64"
     - script: make edge_cross_build
-      name: "cross build edge"     # names the fifth Tests stage job
+      name: "cross build edge"
     - script: make edge_cross_build_v7
-      name: "cross build edge for armv7"     # names the fifth Tests stage job
+      name: "cross build edge for armv7"
     - script: make edgesite_cross_build
-      name: "cross build edgesite"    # names the fifth Tests stage job
+      name: "cross build edgesite"
     - script: make edgesite_cross_build_v7
-      name: "cross build edgeitefor armv7"     # names the fifth Tests stage job
+      name: "cross build edgeite for armv7"
 
     - stage: "unit tests"
       script: make edge_test
-      name: "unit test edge"     # names the third Tests stage job
+      arch: amd64
+      name: "unit test edge on amd64"
+      #- script: make edge_test
+      #- arch: arm64
+      #- name: "unit test edge on arm64"
     - script: make edge_integration_test
-      name: "integration test edge"     # names the fourth Tests stage job
+      arch: amd64
+      name: "integration test edgeo on amd64"
+      #- script: make edge_integration_test
+      #- arch: arm64
+      #- name: "integration test edge on arm64"
+    - script: make cloud_test
+      arch: amd64
+      name: "unit test cloud on amd64"
     - script: make cloud_test
-      name: "unit test cloud" 
+      arch: arm64
+      name: "unit test cloud on arm64"
```

Comment:
do we need unit tests on both architectures? I mean the unit test will test if we have made failures on the code and not if it works on this architecture.

Reply:
agreed, but we need make sure all the dependencies needed for the unit test also passed on the ARM, i.e. we need to be able to run those unit test on ARM as well.